pkg/reduce/readloop/readloop.go

```diff
@@ -102,6 +102,8 @@ func (rl *ReadLoop) Startup(ctx context.Context) error {
 		// crosses the window.
 
 		alignedKeyedWindow := keyed.NewKeyedWindow(p.Start, p.End)
+		// add key to the window
+		alignedKeyedWindow.AddKey(p.Key)
```

Comment:
how was it working before?

Reply:
@yhl25 please update the comment to explain why we are doing it

Reply:
Okayy